skrub/_gap_encoder.py

```diff
@@ -198,6 +199,57 @@ def _init_w(self, V: NDArray, X) -> tuple[NDArray, NDArray, NDArray]:
         B = A.copy()
         return W, A, B
 
+    def _minibatch_convergence(self, batch_size, batch_cost, n_samples, step, n_steps):
+        """Helper function to encapsulate the early stopping logic"""
+        # adapted from sklearn.decomposition.MiniBatchNMF
+
+        # counts steps starting from 1 for user friendly verbose mode.
+        step = step + 1
+
+        # Ignore first iteration because H is not updated yet.
+        if step == 1:
+            if self.verbose:
+                print(f"Minibatch step {step}/{n_steps}: mean batch cost: {batch_cost}")
+            return False
+
+        # Compute an Exponentially Weighted Average of the cost function to
+        # monitor the convergence while discarding minibatch-local stochastic
+        # variability: https://en.wikipedia.org/wiki/Moving_average
```

Comment:
Same comment as #681, keeping the naming the same as Wikipedia could help with maintenance:
```suggestion
        # Compute an Exponentially Moving Average (EMA) of the cost function to
        # monitor the convergence while discarding minibatch-local stochastic
        # variability: https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average
```
If accepted, the variables need to be renamed accordingly (e.g. `_ewa_cost` to `_ema_cost`).

Reply:
Hum I think I prefer to keep the same name than in sklearn's MinibatchNMF, WDYT? The full name is Exponential Weighted Moving Average so both ewa and ema seem fine to me.